app/scripts/metamask-controller.js

```diff
@@ -510,6 +510,15 @@ export default class MetamaskController extends EventEmitter {
           listener,
         ),
     });
+    this.controllerMessenger.registerActionHandler(
+      `PreferencesController:getState`,
+      () => this.preferencesController.store.getState(),
+    );
+    this.controllerMessenger.registerInitialEventPayload({
+      eventType: `PreferencesController:stateChange`,
+      getPayload: () => [this.preferencesController.store.getState(), []],
+    });
+    this.controllerMessenger.subscribe(`PreferencesController:stateChange`);
```

Comment:
These will become redundant once the extension preferences-controller is replaced with the core repo `@metamask/preferences-controller`. 

In the meantime, they are necessary because `TokenDetectionController` and `TokensController` have been upgraded to communicate with the `PreferencesController` using the controller-messenger pattern, instead of passing in callbacks or the entire controller into their constructors.

Reply:
```suggestion
	// TODO: Remove once `PreferencesController` is migrated to the core repo
    this.controllerMessenger.registerActionHandler(
      `PreferencesController:getState`,
      () => this.preferencesController.store.getState(),
    );
    // TODO: Remove once `PreferencesController` is migrated to the core repo
    this.controllerMessenger.registerInitialEventPayload({
      eventType: `PreferencesController:stateChange`,
      getPayload: () => [this.preferencesController.store.getState(), []],
    });
    // TODO: Remove once `PreferencesController` is migrated to the core repo
    this.controllerMessenger.subscribe(`PreferencesController:stateChange`);
```